packages/nimble-components/src/table/specs/table-row-grouping-hld.md

```diff
@@ -0,0 +1,228 @@
+# Row Grouping HLD
```

Comment:
General question about grouping:
Is there any coordination that should be done by the nimble-table for grouping + sorting. For example, should grouped columns always be sorted first so that the groups are visualized in a deterministic order?

Reply:
I think we might be talking about two different things -- it sounds like you are talking about column ordering, but I was talking about column sorting. Take something like an "Age" column as an example, if you group by that column, should the groups always be displayed as "0", "1", "2", etc? By default, they will just be grouped in the order it finds values in the records which could lead to something like "5", "1", "2", "0", etc, which seems confusing.

So, the question is whether all of the groups should implicitly be sorted by the table?

Reply:
This was marked as an open question in the [IxD wireframes](https://xd.adobe.com/view/6f3be15d-8503-4f1f-54b9-5bc27955b3e4-190a/screen/5c0df4b8-a2ab-485b-a322-78ec1ff58c90) (page 25).

Personally, I think group rows should always be displayed in an ascending order (while the children inside those groups should follow the table sort order), so groups are always listed A-Z, 0-99, etc.

Reply:
@atmgrifter00 Sure. The Ag-grid example seems the most straight-forward, and I'm okay if "group by" on an unsorted column displays the groups in an unsorted order until the user sorts the column by ascending or descending.